src/System.Management.Automation/engine/parser/SemanticChecks.cs

```diff
@@ -933,7 +942,7 @@ public override AstVisitAction VisitConvertExpression(ConvertExpressionAst conve
                     if (childAttrExpr != null)
                     {
                         var childConvert = childAttrExpr as ConvertExpressionAst;
-                        if (childConvert != null && typeof(PSReference) == childConvert.Type.TypeName.GetReflectionType())
+                        if (childConvert != null && !IsOrderedAttribute(childConvert.Type) && typeof(PSReference) == childConvert.Type.TypeName.GetReflectionType())
```

Comment:
Let's not make any changes here. We are not trying to optimize all cases where `[ordered]xxx` may appear, but the most common ones, which are the cases of simply `[ordered]<expr>` -- it is already covered above.